### ISSUE TITLE
regression: systemd 249.3 cannot add interface to bridge on boot

### ISSUE THREAD

**Issue by p5n** [systemd/systemd#20487]
**systemd version the issue has been seen with**
 > after update 248.3-2 -> 249.3-1

**Used distribution**
 > ArchLinux

**Linux kernel version used** (`uname -a`)
 > `5.13.10-arch1-1`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > lan interface added to brlan bridge

**Unexpected behaviour you saw**
 > It adds it if I delete bridge and restart systemd-networkd. On boot it does not add it.

**Steps to reproduce the problem**
```
$ cat /etc/udev/rules.d/10-network.rules 
SUBSYSTEM=="net", ATTRS{address}=="e0:69:95:50:79:f5", NAME="lan"

$ cat /etc/systemd/network/brlan.netdev 
[NetDev]
Name=brlan
Kind=bridge
[Bridge]
STP=yes

$ cat /etc/systemd/network/lan.network 
[Match]
Name=lan
[Network]
Bridge=brlan
```

**Comment by p5n** [comment]
I was able to work it around by removing interface renaming and adding it to bridge by MAC.
I believe it related to:
```
systemd-networkd[577]: lan: Interface name change detected, renamed to eth1.
systemd-networkd[577]: Could not process link message: File exists
systemd-networkd[577]: eth1: Failed
systemd-networkd[577]: eth1: Interface name change detected, renamed to lan.
```
Machine has 4 cards: eth0, eth1, lan, test

**Comment by yuwata** [comment]
Sounds duplicate of #20203, which is already fixed by 176b8be10ffce2f8c1fc931a37904a528057016f. The commit is already backported to v249-stable repository, but not included in v249.3. Please wait for v249.4 or newer, or apply the commit by your self.

If the commit does not fix your issue, then please request to reopen this. In that case, please attach debugging logs of networkd.

Closing.
